pkg/pipeline/pipeline_test.go

```diff
@@ -75,33 +79,67 @@ pipeline:
 `
 
 func Test_SimplePipeline(t *testing.T) {
-	var json = jsoniter.ConfigCompatibleWithStandardLibrary
-	var mainPipeline *Pipeline
-	var err error
-	var b []byte
-	v := test.InitConfig(t, configTemplate)
-	config.Opt.PipeLine.Ingest.Type = "file"
-	config.Opt.PipeLine.Decode.Type = "json"
-	config.Opt.PipeLine.Extract.Type = "none"
-	config.Opt.PipeLine.Encode.Type = "none"
-	config.Opt.PipeLine.Write.Type = "none"
-	config.Opt.PipeLine.Ingest.File.Filename = "../../hack/examples/ocp-ipfix-flowlogs.json"
+	loadGlobalConfig(t)
 
-	val := v.Get("pipeline.transform\n")
-	b, err = json.Marshal(&val)
-	require.NoError(t, err)
-	config.Opt.PipeLine.Transform = string(b)
-
-	mainPipeline, err = NewPipeline()
+	mainPipeline, err := NewPipeline()
 	require.NoError(t, err)
 
 	// The file ingester reads the entire file, pushes it down the pipeline, and then exits
 	// So we don't need to run it in a separate go-routine
 	mainPipeline.Run()
+
 	// What is there left to check? Check length of saved data of each stage in private structure.
 	ingester := mainPipeline.Ingester.(*ingest.IngestFile)
 	decoder := mainPipeline.Decoder.(*decode.DecodeJson)
 	writer := mainPipeline.Writer.(*write.WriteNone)
+	require.Equal(t, 5103, len(ingester.PrevRecords))
 	require.Equal(t, len(ingester.PrevRecords), len(decoder.PrevRecords))
 	require.Equal(t, len(ingester.PrevRecords), len(writer.PrevRecords))
+
+	// checking that the processing is done for at least the first line of the logs
+	require.Equal(t, ingester.PrevRecords[0], decoder.PrevRecords[0])
+	// values checked from the first line of the ../../hack/examples/ocp-ipfix-flowlogs.json file
+	require.Equal(t, config.GenericMap{
+		"fl2m_bytes":   float64(20800),
+		"fl2m_dstAddr": "10.130.2.2",
+		"fl2m_dstPort": float64(36936),
+		"fl2m_packets": float64(400),
+		"fl2m_srcAddr": "10.130.2.13",
+		"fl2m_srcPort": float64(3100),
+	}, writer.PrevRecords[0])
+}
+
+func BenchmarkPipeline(b *testing.B) {
```

Comment:
@mariomac can you move the `Benchmark` into a separate file so we will be able to run just benchmarks and it will be easy to see that something is benchmarking (in compared to a test). THX

Reply:
ok ... NP